src/core/vdom.ts

```diff
@@ -1609,7 +1648,14 @@ export function renderer(renderer: () => RenderResult): Renderer {
 					setValue(domNode, propValue, previousValue);
 				} else if (propName !== 'key' && propValue !== previousValue) {
 					const type = typeof propValue;
-					if (type === 'function' && propName.lastIndexOf('on', 0) === 0 && includesEventsAndAttributes) {
+					if (type === 'object' && propValue && 'handler' in propValue) {
```

Comment:
Would this also need the `includesEventsAndAttributes` to ensure that events don't get registered twice? Another thing would be that we probably need to add support for this new style of event handlers when using `dom()` function (https://github.com/dojo/framework/blob/master/src/core/interfaces.d.ts#L171)

When support has been added for that then it might make sense to put all the logic handling either a function or object with a handler inside the `updateEvent` function.